src/bgw/job_stat.c

```diff
@@ -368,6 +369,8 @@ calculate_next_start_on_failure(TimestampTz finish_time, int consecutive_failure
 		Interval retry_ival = { .time = 2000000 };
 		retry_ival.time += rand_backoff;
 
+		BeginInternalSubTransaction("next start on failure");
```

Comment:
might be just theory but in case we have this instruction here: I wonder what happens if this command errors out?
* the catch will call `RollbackAndReleaseCurrentSubTransaction`
* but since the open failed that will be affect the parent if there is any...

Reply:
seems like this method internally [does](https://github.com/postgres/postgres/blob/master/src/backend/access/transam/xact.c#L4599-L4600):
```
        CommitTransactionCommand();
        StartTransactionCommand();
```
if we don't have a "subtransaction" already open - wouldn't that will mean to commit the parent txn?

Reply:
I've cleared this method - it sets a flag internally ; and with that `CommitTransaction` behaves differently ; looks good